.github/workflows/external-dependencies-tests.yml

```diff
@@ -0,0 +1,21 @@
+name: External Dependencies Tests
+
+on:
+    pull_request:
+        branches: [main]
```

Comment:
shouldn't we test on staging as well?

Reply:
This gh-action runs the external-dependencies tests directory.
The external dependencies test directory should be limited to only PR to main because we call openai chat completion in there and that has a cost/million token.

I'm looking at this directory to hold tests that requires API calls or dependencies you wouldn't want running too often, and on PR to main seems like a good approach